include/golioth/golioth_sys.h

```diff
@@ -121,6 +121,70 @@ void golioth_sys_thread_destroy(golioth_sys_thread_t thread);
 #define golioth_sys_rand() rand()
 #endif
 
```

Comment:
When #666 merges I think we should guard the following function declarations so they are only available when `CONFIG_GOLIOTH_OTA` is selected.

The same goes for the definitions in each platform file.